### PR TITLE
BOM-2649: Django 3.2 Support

### PR DESCRIPTION
**Pull request by mzulqarnain1**
## Description

- Added Django 3.2 Support
- Applied Django Code Mods & Pyupgrade 
- Ran Make Upgrade

## Ticket Link

Link to the associated ticket
https://openedx.atlassian.net/browse/BOM-2649

### CODE REVIEW

File: enterprise_catalog/apps/catalog/forms.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Meta:
         fields = ('content_filter',)
 
     def validate_content_filter_fields(self, content_filter):
-        for key in cftypes:
+        for key in cftypes:  # pylint: disable=consider-using-dict-items
```
Comment (Jawayria) [inline enterprise_catalog/apps/catalog/forms.py]: I think fixing this warning won't raise any compatibility issues.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       max-parallel: 4
```
Comment (mraarif) [inline .github/workflows/ci.yml]: we should remove this restriction.

File: enterprise_catalog/docker_gunicorn_configuration.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 
 def pre_request(worker, req):
-    worker.log.info("%s %s" % (req.method, req.path))
+    worker.log.info(f"{req.method} {req.path}")
```
Comment (mraarif) [inline enterprise_catalog/docker_gunicorn_configuration.py]: log statements shouldn't use `f` strigns

File: enterprise_catalog/apps/catalog/forms.py (hunk 1 above)
Comment (mzulqarnain1) [reply enterprise_catalog/apps/catalog/forms.py]: Fixed the warning & removed suppression.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mzulqarnain1) [reply .github/workflows/ci.yml]: Done.

File: enterprise_catalog/docker_gunicorn_configuration.py (hunk 3 above)
Comment (mzulqarnain1) [reply enterprise_catalog/docker_gunicorn_configuration.py]: Done.

**Review by mraarif — APPROVED** [review]

**Review by macdiesel — APPROVED** [review]

**Comment by natabene** [comment]
@macdiesel Please merge at your earliest convenience.

**Pull request merged**